bin/zowe-install-component.sh

```diff
@@ -0,0 +1,333 @@
+#!/bin/sh
+#######################################################################
+# This program and the accompanying materials are made available
+# under the terms of the Eclipse Public License v2.0 which
+# accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright Contributors to the Zowe Project. 2020
+#######################################################################
+
+#######################################################################
+# Zowe Component Installer
+#
+# This script will install a component into target directory. The component to
+# be installed can be a pax, zip or tar package, or a directory.
+#
+# Note: this script works better with NODE_HOME. But for backward compatible
+#       purpose, NODE_HOME is not mandatory.
+#
+# Command line options:
+# -c|--component-name optional. component name. If NODE_HOME is defined, and the
+#                     component has manifest, this value is optional. Otherwise
+#                     it's required.
+# -o|--component-file required. path to the component package or directory.
+# -i|--instance-dir   optional. path to Zowe instance directory. If this
+#                     has a value, the script will also execute
+#                     zowe-configure-component.sh.
+# -d|--target-dir     optional. path to target directory. For native component,
+#                     default value is ${ZOWE_ROOT_DIR}/components. For non-
+#                     native component, the script will check ZWE_EXTENSION_DIR
+#                     if possible. Otherwise will fall back to
+#                     ${DEFAULT_TARGET_DIR}.
+# -e|--auto-encoding  optional. If we want to automatically tagging the
+#                     directory. Default value is "auto". Allowed values are:
+#                     - yes: automatically tag the files encoding
+#                     - no: do not automatically tag encoding
+#                     - auto: only tag when manifest is in ISO8859-1 encoding.
+#                     This option is only applicable for z/OS.
+# -k|--core           optional boolean. Whether this component is bundled
+#                     into Zowe package.
+# -l|--logs-dir       optional. path to logs directory.
+# -f|--log-file       optional. write log to the file specified.
+#######################################################################
+
+#######################################################################
+# Constants
+DEFAULT_TARGET_DIR=/global/zowe/extensions
+
+#######################################################################
+# Prepare shell environment
+if [ -z "${ZOWE_ROOT_DIR}" ]; then
+  export ZOWE_ROOT_DIR=$(cd $(dirname $0)/../;pwd)
+fi
+
```

Comment:
I wonder if we should call {RUNTIME_DIR}/bin/internal/zowe-set-env.sh if this might be called by an extension in order to ensure out settings are consistent?